File: auth2/auth2-demo/src/main/java/com/cwd/auth2/service/UserService.java
package com.cwd.auth2.service;

import com.cwd.auth2.config.auth2.authorization.EHRJdbcClientDetailsService;
import com.cwd.auth2.dao.UserDao;
import com.cwd.auth2.model.User;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.security.oauth2.provider.ClientDetails;
import org.springframework.security.oauth2.provider.NoSuchClientException;
import org.springframework.stereotype.Service;

/**
 * Created by Administrator on 2017/1/11.
 */
@Service
public class UserService implements UserDetailsService {
    @Autowired
    private EHRJdbcClientDetailsService clientDetailsService;
    @Autowired
    private UserDao userDao;

    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
        ClientDetails clientDetails;
        try {
            User user = userDao.findByUsername(username);
            clientDetails = clientDetailsService.loadClientByClientId(user.getClientId());
        } catch (NoSuchClientException e) {
            throw new UsernameNotFoundException(e.getMessage(), e);
        }
        String clientSecret = clientDetails.getClientSecret();
        return new org.springframework.security.core.userdetails.User(username, clientSecret, clientDetails.getAuthorities());
    }
}

File: sms/sms-bootstrap/src/main/scala/com/yihu/cwd/service/UserService.scala
package com.yihu.cwd.service

import com.yihu.cwd.dao.UserDao
import com.yihu.cwd.model.User
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

/**
  * Created by chenweida on 2017/3/28.
  */
@Service
class UserService {


  @Autowired
  private var userDao: UserDao = _;


  def findUserByLoginName(username: String): User = {
    userDao.findByLoginName(username)
  }


  def findUserById(userId: String) = {
    userDao.findById(userId);
  }
}
